src/MoBi.Core/Services/SimulationConfigurationFactory.cs

```diff
@@ -0,0 +1,40 @@
+﻿using MoBi.Core.Domain.Model;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Domain.Formulas;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Utility.Extensions;
+
+namespace MoBi.Core.Services
+{
+   public interface ISimulationConfigurationFactory
+   {
+      SimulationConfiguration Create();
+   }
+   
+   public class SimulationConfigurationFactory : ISimulationConfigurationFactory
+   {
+      private readonly ICoreCalculationMethodRepository _calculationMethodRepository;
+      private readonly ICloneManagerForBuildingBlock _cloneManager;
+      private readonly IMoBiContext _context;
+
+      public SimulationConfigurationFactory(ICoreCalculationMethodRepository calculationMethodRepository, ICloneManagerForBuildingBlock cloneManager, IMoBiContext context)
+      {
+         _calculationMethodRepository = calculationMethodRepository;
+         _cloneManager = cloneManager;
+         _context = context;
+      }
+
+      public SimulationConfiguration Create()
+      {
+         var simulationConfiguration = new SimulationConfiguration
+         {
+            SimulationSettings = _cloneManager.Clone(_context.CurrentProject.SimulationSettings)
```

Comment:
Can those be null? They should not right?

Reply:
you mean ```_context.CurrentProject.SimulationSettings```. No that should not be null. If that happens to you again then try to write the steps.